### PR TITLE
Data submission wf

### PR DESCRIPTION
**Pull request by lindaxiang**

### CODE REVIEW

File: docs/submission/submitting-molecular-data.md [hunk 1]
```diff
@@ -12,161 +15,688 @@ Raw molecular data is submitted to a **Regional Data Processing Centre (RDPC)**.
...
-Download the **[latest version of the song-client](https://artifacts.oicr.on.ca/artifactory/dcc-release/bio/overture/song-client/[RELEASE]/song-client-[RELEASE]-dist.tar.gz)**. Once you have unzipped the tarball, change directories into the unzipped folder:
+### Workflow Diagram
 
-```shell
-wget -O song-client.tar.gz https://artifacts.oicr.on.ca/artifactory/dcc-release/bio/overture/song-client/[RELEASE]/song-client-[RELEASE]-dist.tar.gz
+![workflow-digram](/assets/submission/ICGC-ARGO-data-submission-wf.png)
+
+### Workflow Summary
+
+By default, it is assumed that Sequencing Reads are submitted from local storage. However if the Sequencing Reads are stored remotely in EGA Archives, it is easy to enable EGA download sub-workflow to support getting data from EGA and submit to RDPC. The workflow also adds support to Sequencing Reads in CRAM format. For this, the reference genome is required to enable cram2bam step. Currently the workflow performs the following steps:
```
Comment (edsu7) [inline docs/submission/submitting-molecular-data.md:30]: Formating edit:
By default, it is assumed that Sequencing Reads are submitted from local storage. However if the Sequencing Reads are stored remotely in EGA Archives, it is easy to enable EGA download sub-workflow to support getting data from EGA and submit to RDPC. The workflow also adds support to Sequencing Reads in CRAM format. For this, the reference genome is required to enable `cram2bam` step. Currently the workflow performs the following steps:

File: docs/submission/submitting-molecular-data.md [hunk 2]
```diff
@@ -12,161 +15,688 @@ Raw molecular data is submitted to a **Regional Data Processing Centre (RDPC)**.
...
+[-        ] process > ArgoDataSubmissionWf:submissionReceipt   -
+```
+
+Please verify if you’ve provided the required params to invoke the expected processes.
+
+4. Start submitting your own data following the steps in section [How to Upload Molecular Data](/docs/submission/submitting-molecular-data#how-to-upload-molecular-data)
+
+## How to Upload Molecular Data
+
+Molecular data files are submitted in conjunction with necessary, descriptive data about molecular files and samples, called metadata. The DCC requires that metadata is submitted at the same time as the molecular data to facilitate automated downstream analysis. This metadata is requested to be one of the mandatory inputs for the data submission workflow. Before proceeding, please read the instructions on how to prepare and format the metadata according to [molecular metadata model](/docs/submission/submitting-molecular-data#understanding-the-metadata-model) and [validation rules](/docs/submission/submitting-molecular-data#molecular-data-and-metadata-validation-rules).
```
Comment (edsu7) [inline docs/submission/submitting-molecular-data.md:74]: Molecular data files are submitted in conjunction with necessary, descriptive data about molecular files and samples, called metadata. The DCC requires metadata to be submitted at the same time as the molecular data to facilitate automated downstream analysis. The metadata is one of the mandatory inputs for the data submission workflow. Before proceeding, please read the instructions on how to prepare and format the metadata according to [molecular metadata model](/docs/submission/submitting-molecular-data#understanding-the-metadata-model) and [validation rules](/docs/submission/submitting-molecular-data#molecular-data-and-metadata-validation-rules).

File: docs/submission/submitting-molecular-data.md [hunk 3]
```diff
@@ -12,161 +15,688 @@ Raw molecular data is submitted to a **Regional Data Processing Centre (RDPC)**.
...
+The first step of submitting molecular data is to prepare the metadata TSV files conforming to the most recent [metadata dictionary](/docs/submission/submitting-molecular-data#metadata-dictionary) that has been defined by the DCC.
+
+The ARGO Data Platform currently accepts the following types of genomic sequencing analyses:
+
+- DNA-Seq data from tumour and normal paired Whole Genome Sequencing (WGS) and Whole Exome Sequencing (WXS)
+- RNA-Seq data from Transcriptome Sequencing (RNA-Seq)
+
+There are submissions for either Tumour or Normal samples, from Paired-end vs. Single-end sequencing, in either BAM, FASTQ or CRAM formats. The data submission workflow can be configured to submit either local data or remote data stored in EGA archives. Each of these scenarios has a specific set of metadata rules to follow.
+
+#### experiment
```
Comment (edsu7) [inline docs/submission/submitting-molecular-data.md:87]: #### Experiment

File: docs/submission/submitting-molecular-data.md [hunk 4]
```diff
@@ -12,161 +15,688 @@ Raw molecular data is submitted to a **Regional Data Processing Centre (RDPC)**.
...
+
+- Fields `program_id`, `submitter_[donor|specimen|sample]_id` are `Required` and must be the same as what are submitted in `sample_registration` table.
+- Field `submitter_matched_normal_sample_id` is `Required` for `WGS/WXS` tumour samples.
+- Fields `submitter_sequencing_experiment_id`, `experimental_strategy`, `platform` and `read_group_count` are `Required` for all samples.
+- Fields `target_capture_kit`, `primary_target_regions` and `capture_target_regions` are `Required` for `Targeted-Seq/WXS` samples.
+- Field `library_strandedness` is `Required` for `RNA-Seq` samples.
+
+:::
+
+#### read_groups
```
Comment (edsu7) [inline docs/submission/submitting-molecular-data.md:181]:
```
#### Read_groups
```

File: docs/submission/submitting-molecular-data.md [hunk 5]
```diff
@@ -12,161 +15,688 @@ Raw molecular data is submitted to a **Regional Data Processing Centre (RDPC)**.
...
+:::note
+
+- Fields `submitter_sequencing_experiment_id`, `submitter_read_group_id`, `platform_unit`, `is_paired_end`, and `library_name` are `Required` for all read_groups.
+- Field `read_group_id_in_bam` is `Required` for `BAM/CRAM` files, however this can NOT be submitted for FASTQ files.
+- Field `file_r1` is `Required`, and both `file_r1` and `file_r2` must match a fileName identified in the `files` section.
+- Fields `file_r2`, `read_length_r2` and `insert_size` are `Required` if and only if paired-end sequencing was done.
+
+:::
+
+#### files
```
Comment (edsu7) [inline docs/submission/submitting-molecular-data.md:247]:
```
#### Files
```

File: docs/submission/submitting-molecular-data.md [hunk 6]
```diff
@@ -12,161 +15,688 @@ Raw molecular data is submitted to a **Regional Data Processing Centre (RDPC)**.
...
+
+The molecular metadata model is represented as a graph with nodes and edges. It maintains the critical relationship between entities including program, donor, specimen, sample, experiment, read groups and files. The node/edge structure is depicted below:
+
+![metadata-model](/assets/submission/molecular-metadata-model-1.png)
+
+Each blue node represents an entity, edges between nodes represent relationships between entities. There is a lot of metadata associated with entities, which are represented by yellow nodes.
+
+### Metadata Dictionary
+
+The following describes the attributes and permissible values for all of the fields within the metadata tsv files for the ARGO Data Platform and Molecuar Data Processing.
```
Comment (hknahal) [inline docs/submission/submitting-molecular-data.md:457]: Minor spelling typo: Molecu`l`ar

File: docs/submission/submitting-molecular-data.md [hunk 7]
```diff
@@ -12,161 +15,738 @@ Raw molecular data is submitted to a **Regional Data Processing Centre (RDPC)**.
...
+
+Molecular data files are submitted in conjunction with necessary, descriptive data about molecular files and samples, called metadata. The DCC requires that metadata is submitted at the same time as the molecular data to facilitate automated downstream analysis. This metadata is one of the mandatory inputs for the data submission workflow. Before proceeding, please read the instructions on how to prepare and format the metadata according to [molecular metadata model](/docs/submission/submitting-molecular-data#understanding-the-metadata-model) and [validation rules](/docs/submission/submitting-molecular-data#molecular-data-and-metadata-validation-rules).
+
+### Step 1. Prepare Molecular Metadata
+
+The first step of submitting molecular data is to prepare the metadata TSV files conforming to the most recent [metadata dictionary](/docs/submission/submitting-molecular-data#metadata-dictionary) that has been defined by the DCC.
+
+The ARGO Data Platform currently accepts the following types of genomic sequencing analyses:
+
+- DNA-Seq data from tumour and normal paired Whole Genome Sequencing (WGS) and Whole Exome Sequencing (WXS)
```
Comment (edsu7) [inline docs/submission/submitting-molecular-data.md:82]: should we add target sequencing?

File: docs/submission/submitting-molecular-data.md [hunk 8]
```diff
@@ -12,161 +15,738 @@ Raw molecular data is submitted to a **Regional Data Processing Centre (RDPC)**.
...
+
+</TabItem>
+
+</Tabs>
+
+<!---  Tabs end here -->
+
+:::note
+
+- Fields `program_id`, `submitter_[donor|specimen|sample]_id` are `Required` and must be the same as what are submitted in `sample_registration` table.
```
Comment (edsu7) [inline docs/submission/submitting-molecular-data.md:173]:
```
Fields `program_id`, `submitter_[donor|specimen|sample]_id` are `Required` and must be the same values as the `sample_registration` table.
```

**Review by edsu7 — CHANGES_REQUESTED** [review]

**Review by edsu7 — DISMISSED** [review]
lgtm!

**Review by justincorrigible — DISMISSED** [review]
LGTM 👍

File: website/sidebars.js [hunk 9]
```diff
@@ -31,7 +31,6 @@ module.exports = {
         'submission/clinical-data-validation-rules',
         'submission/submitting-clinical-data',
         'submission/viewing-clinical-data',
-        'submission/molecular-data-prep',
```
Comment (justincorrigible) [inline website/sidebars.js:34]: not sure how does this affect the menu... with the filename changes in the two .md files in this PR.
guess we'll find out soon enough

**Pull request merged**